drivers/flash/soc_flash_nrf_ticker.c

```diff
@@ -0,0 +1,224 @@
+				     /* flash work */
+			   context, /* the context for the flash operation */
+			   NULL, /* no op callback */
+			   NULL);
+
+	if (err != TICKER_STATUS_SUCCESS && err != TICKER_STATUS_BUSY) {
+		_ticker_synch_context.result = 0;
+
+		/* abort flash timeslots */
+		_ticker_stop(instance_index, 0, ticker_id);
+
+		/* notify thread that data is available */
+		k_sem_give(&sem_sync);
+	}
+}
+
+static void time_slot_callback_abort(uint32_t ticks_at_expire,
+				     uint32_t remainder,
+				     uint16_t lazy, void *context)
+{
+	ll_radio_state_abort();
+	time_slot_delay(ticks_at_expire,
+			HAL_TICKER_US_TO_TICKS(FLASH_RADIO_WORK_DELAY_US),
+			time_slot_callback_work,
+			context);
+}
+
+static void time_slot_callback_prepare(uint32_t ticks_at_expire,
+				       uint32_t remainder,
+				       uint16_t lazy, void *context)
+{
+#if defined(CONFIG_BT_CTLR_LOW_LAT)
+	time_slot_callback_abort(ticks_at_expire, remainder, lazy, context);
+#else /* !CONFIG_BT_CTLR_LOW_LAT */
+	time_slot_delay(ticks_at_expire,
+			HAL_TICKER_US_TO_TICKS(FLASH_RADIO_ABORT_DELAY_US),
+			time_slot_callback_abort,
+			context);
+#endif /* CONFIG_BT_CTLR_LOW_LAT */
+}
+
+
+int nrfsb_synch_backend_init(void)
+{
+	k_sem_init(&sem_sync, 0, 1);
+	return 0;
+}
+
+void nrfsb_set_synch_context(uint32_t duration)
+{
+
+	/* FLASH_SYNCH_SWITCHING_TIME is delay which is allways added by
+	 * the slot calling mechanism
+	 */
+	_ticker_synch_context.interval = duration - FLASH_SYNCH_SWITCHING_TIME;
+	_ticker_synch_context.slot = duration;
+}
+
+int nrfsb_work_synch_with_radio(struct flash_op_desc *op_desc)
+{
+	uint8_t instance_index;
+	uint8_t ticker_id;
+	int result;
+	uint32_t err;
+
+	ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
+
+	err = ticker_start(instance_index,
+			   3, /* user id for thread mode */
+			      /* (MAYFLY_CALL_ID_PROGRAM) */
+			   ticker_id, /* flash ticker id */
+			   ticker_ticks_now_get(), /* current tick */
+			   0, /* first int. immediately */
+			   /* period */
+			   HAL_TICKER_US_TO_TICKS(
+				_ticker_synch_context.interval),
+			   /* period remainder */
+			   HAL_TICKER_REMAINDER(_ticker_synch_context.interval),
+			   0, /* lazy, voluntary skips */
+			   HAL_TICKER_US_TO_TICKS(_ticker_synch_context.slot),
+			   time_slot_callback_prepare,
+			   op_desc,
+			   NULL, /* no op callback */
+			   NULL);
+
+	if (err != TICKER_STATUS_SUCCESS && err != TICKER_STATUS_BUSY) {
+		result = -ECANCELED;
+	} else if (k_sem_take(&sem_sync, K_MSEC(FLASH_TIMEOUT_MS)) != 0) {
+		/* Stop any scheduled jobs */
+		_ticker_stop(instance_index, 3, ticker_id);
+
+		/* wait for operation's complete overrun*/
+		result = -ETIMEDOUT;
+	} else {
+		result = _ticker_synch_context.result;
+	}
+
+	return result;
+}
+
+bool nrfsb_is_synch_required(void)
+{
+	return ticker_is_initialized(0);
+}
+
+void nrfsb_get_timestamp_begin(void)
+{
+	_ticker_synch_context.ticks_begin = ticker_ticks_now_get();
+}
```

Comment:
Could these be given `inline` specifier?

Reply:
? this is API implementation, can't be inline.

Reply:
or can be but it don nothing without static...
and can't be static for the other reason...

Reply:
I think that it can be `static inline` in header. Have seen this somewhere in Zephyr headers.
Drawback is that if compiler fails to inline function it adds another callable static to object.